src/domain/interfaces/counterfactual-safes.datasource.interface.ts

```diff
@@ -7,25 +7,27 @@ export const ICounterfactualSafesDatasource = Symbol(
 );
 
 export interface ICounterfactualSafesDatasource {
-  createCounterfactualSafe(
-    account: Account,
-    createCounterfactualSafeDto: CreateCounterfactualSafeDto,
-  ): Promise<CounterfactualSafe>;
+  createCounterfactualSafe(args: {
+    account: Account;
+    createCounterfactualSafeDto: CreateCounterfactualSafeDto;
+  }): Promise<CounterfactualSafe>;
 
-  getCounterfactualSafe(
-    chainId: string,
-    predictedAddress: `0x${string}`,
-  ): Promise<CounterfactualSafe>;
+  getCounterfactualSafe(args: {
+    chainId: string;
+    predictedAddress: `0x${string}`;
+  }): Promise<CounterfactualSafe>;
 
-  getCounterfactualSafesForAccount(
-    account: Account,
-  ): Promise<CounterfactualSafe[]>;
+  getCounterfactualSafesForAccount(args: {
+    account: Account;
+  }): Promise<CounterfactualSafe[]>;
 
-  deleteCounterfactualSafe(
-    account: Account,
-    chainId: string,
-    predictedAddress: `0x${string}`,
-  ): Promise<void>;
+  deleteCounterfactualSafe(args: {
+    account: Account;
+    chainId: string;
+    predictedAddress: `0x${string}`;
+  }): Promise<void>;
 
-  deleteCounterfactualSafesForAccount(account: Account): Promise<void>;
+  deleteCounterfactualSafesForAccount(args: {
+    account: Account;
+  }): Promise<void>;
```

Comment:
Nit: we don't use objects for singular arguments elsewhere (`getCounterfactualSafesForAccount` and `deleteCounterfactualSafesForAccount`). Do you intend to extend them?

Reply:
I don't have an strong opinion, maybe it's better to stick to the standards we have been following and use objects only for non-singular arguments. Will change that, thanks!

Reply:
I've removed the types for single arguments in [1d0832e](https://github.com/safe-global/safe-client-gateway/pull/1776/commits/1d0832eba1b1e72bdf25fb974ce02f7cc71da0a3)